amd/src/expand-topics.js

```diff
@@ -1,45 +1,42 @@
-const waitForElements = () => {
-    const courseIndexButtons = document.querySelectorAll('.courseindex-section');
-
-    if (courseIndexButtons.length > 0) {
-        // Add listeners.
-        courseIndexButtons.forEach(function (button) {
-            button.addEventListener('click', handleButtonClick);
-        });
-    } else {
-        // Elements not loaded yet.
-        setTimeout(waitForElements, 250);
-    }
-};
-
 const handleButtonClick = function (event) {
     event.preventDefault();
 
-    // Extract the section id from the URL.
-    const url = this.href;
-    const sectionNumber = url.match(/#section-(\d+)/)[1];
+    const button = event.target.closest('.courseindex-section');
 
-    // Expand relevant section if needed.
-    const courseContentCollapse = document.getElementById(`coursecontentcollapse${sectionNumber}`);
-    const collapseSection = document.getElementById(`collapssesection${sectionNumber}`);
+    if (button) {
+        const url = button.href;
+        const sectionNumber = url.match(/#section-(\d+)/)[1];
 
-    if (courseContentCollapse && !courseContentCollapse.classList.contains('show')) {
-        courseContentCollapse.classList.add('show');
-    }
+        // Expand relevant section if needed.
+        const courseContentCollapse = document.getElementById(`coursecontentcollapse${sectionNumber}`);
+        const collapseSection = document.getElementById(`collapssesection${sectionNumber}`);
+
+        if (courseContentCollapse && !courseContentCollapse.classList.contains('show')) {
+            courseContentCollapse.classList.add('show');
+        }
 
-    if (collapseSection && collapseSection.classList.contains('collapsed')) {
-        collapseSection.classList.remove('collapsed');
+        if (collapseSection && collapseSection.classList.contains('collapsed')) {
+            collapseSection.classList.remove('collapsed');
+        }
+
+        // Go to section.
+        const courseSection = document.getElementById(`section-${sectionNumber}`);
+        courseSection.scrollIntoView({ behavior: 'smooth' });
     }
+};
 
-    // Go to section.
-    const courseSection = document.getElementById(`section-${sectionNumber}`);
-    courseSection.scrollIntoView({ behavior: 'smooth' });
+const courseIndexEvent = () => {
+    const courseIndex = document.getElementById('courseindex-content');
+
+    if (courseIndex) {
```

Comment:
Are there any cases where the course index element is dynamically added to the page or is it added statically via backend?

Reply:
Statically added.